test/unit/index.js

```diff
@@ -0,0 +1,145 @@
+'use strict'
+
+const {test, threw} = require('tap')
+
+const config = require('../../lib/config.js')
+const {handler} = require('../../index.js')
+const {s3} = require('../../lib/transformer.js')
+const {setProperty} = require('../../lib/utils.js')
+
+const BUCKET_NAME = 'random_name'
+const ERROR_MESSAGE = 'A connection-based error occurred that will not be retried.'
+  + ' See meta data for details.'
+const ERROR_STATUS_CODE = 403
+const FILE_NAME = 'log+File.json'
+const LOG_LINE = 'test log'
+const TIMESTAMP = Date.now()
+const EVENT_DATA = {
+  Records: [{
+    eventTime: new Date(TIMESTAMP)
+  , s3: {
+      bucket: {
+        name: BUCKET_NAME
+      }
+    , object: {
+        key: FILE_NAME
+      }
+    }
+  }]
+}
+
+// NOTE: We will NOT test the use of ENV vars directly here since they are cumbersome
+// to work with (they can never be updated). For this, we will trust that env-config
+// will do its job and read from the envrionment properly, but we will set the
+// config's keys manually in these tests.
+config.validateEnvVars()
+config.set('ingestion-key', 'required-for-anything-to-work')
```

Comment:
`ingestion-key` should be required in `config.js`, and then swap these two lines (or use `process.env.INGESTION_KEY = 'abc213'` at the top)